lighthouse-core/report/v2/renderer/details-renderer.js

```diff
@@ -134,3 +197,21 @@ DetailsRenderer.ListDetailsJSON; // eslint-disable-line no-unused-expressions
  * }}
  */
 DetailsRenderer.CardsDetailsJSON; // eslint-disable-line no-unused-expressions
+
+/** @typedef {{
+ *     type: string,
+ *     header: ({text: string}|undefined),
+ *     items: !Array<!Array<!DetailsRenderer.DetailsJSON>>,
+ *     itemHeaders: !Array<!DetailsRenderer.DetailsJSON>
```

Comment:
> we're basically descending into recursive madness again

are we though?


`detailsjson` only has type and text.. it can't contain other types.
really it and `thumbnail` should include the word "item" as they are ALWAYS leaf nodes.

and the `render()` method should accept all of these leaf & group types. the real bug seems that render's param is typed to a leaf node when we know its not.